File: examples/ts/code/v1beta/list_engines.ts
import fetch from 'node-fetch'

const apiHost = process.env.API_HOST ?? 'https://api.stability.ai'
const url = `${apiHost}/v1beta/engines/list`

const apiKey = process.env.STABILITY_API_KEY
if (!apiKey) throw new Error('Missing Stability API key.')

const response = await fetch(url, {
  method: 'GET',
  headers: {
    Authorization: `Bearer ${apiKey}`,
  },
})

if (!response.ok) {
  throw new Error(`Non-200 response: ${await response.text()}`)
}

interface Payload {
  engines: Array<{
    id: string
    name: string
    description: string
    type: string
  }>
}

// Do something with the payload...
const payload = (await response.json()) as Payload

File: examples/python/code/v1alpha/list_engines.py
import os
import requests

api_host = os.getenv('API_HOST', 'https://api.stability.ai')
url = f"{api_host}/v1alpha/engines/list"

apiKey = os.getenv("STABILITY_API_KEY")
if apiKey is None:
    raise Exception("Missing Stability API key.")

response = requests.get(url, headers={
    "Authorization": apiKey
})

if response.status_code != 200:
    raise Exception("Non-200 response: " + str(response.text))

# Do something with the payload...
payload = response.json()

